Donation-platform/src/main/java/com/google/PersonalizedRecommendations.java

```diff
@@ -60,37 +63,45 @@
   // Returns mapping of charities to the scores that determine their rank on the personalzied page
   // (higher score = better match)
   public HashMap<Charity, Double> getCharityScores(List<String> selectedTags) {
+    DbSetUpUtils setUp = new DbSetUpUtils();
+    db = setUp.getDbCalls();
+    //setUp.populateDatabase();                                      //only call once
     Collection<Charity> charities = getAllCharities(); // DEFAULT
     // Collection<Charity> charities = getAllHardCodedCharities(); //HARDCODED
     HashMap<Charity, Double> charityScores = new HashMap<Charity, Double>();
 
     for (Charity charity : charities) {
       // If the charity contains any of the user-selected tags, it will appear on the personalized
       // page
-      if ((charity.getTags()).contains(selectedTags.get(0))
-          | (charity.getTags()).contains(selectedTags.get(1))
-          | (charity.getTags()).contains(selectedTags.get(2))) {
+      Collection<Tag> charityTagObjects = charity.getCategories();
+      Collection<String> charityTagStrings = new ArrayList<String>();
```

Comment:
does this contain tagNames for a charity? If so rename  it to "tagNamesForACharity"

Reply:
Names can be as descriptive as possible but should be meaningful

Reply:
Yes. Fixed.

Reply:
I renamed it "charityTagNames."